File: solutions/clojure/leap/1/src/leap.clj
(ns leap)

(defn rem-0? [num div]
  (= (rem num div) 0))

(defn leap-year? [year]
  (or (rem-0? year 400)
      (and (not (rem-0? year 100))
           (rem-0? year 4))))

File: solutions/c/leap/1/src/leap.c
#include "leap.h"

bool is_leap_year(int year) {
  bool multiple_of_4 = year % 4 == 0;
  bool multiple_of_100 = year % 100 == 0;
  bool multiple_of_400 = year % 400 == 0;

  if (multiple_of_400)
    return true;
  else if (multiple_of_100)
    return false;
  return multiple_of_4;
}